### ISSUE TITLE
Question about Image size and ratio

### ISSUE THREAD

**Issue by peterwilson3912** [sairajk/PyTorch-Pyramid-Feature-Attention-Network-for-Saliency-Detection#13]
Hi! Thank you so much for this amazing work. It's quite helpful. I met a problem right now and need your help. I want to resize the image and make it a rectangle.  
Here 

`parser.add_argument('--img_size', default=256, help='Image size to be used', [type=int)]`(https://github.com/sairajk/PyTorch-Pyramid-Feature-Attention-Network-for-Saliency-Detection/blob/master/inference.py#L23)

I'm wondering what's the largest size can this number be? I changed it to 1280 but there seem problems. And then I use 640 and I found the saliency area changed comparing with 256.

I'm also wondering whether I can change the output image as a rectangle. (16:9) ratio. 

I don't want to resize by some image processing tools because I think it will change the saliency value for each pixel and destroy the one-to-one corresponding pixel relationship. 

I'm wondering do we have solution inside the code.

Thank you so much!

**Comment by sairajk** [comment]
Hi @karliell, 

> I'm wondering what's the largest size can this number be? I changed it to 1280 but there seem problems. And then I use 640 and I found the saliency area changed comparing with 256.

The model is trained on image size (256, 256). So it has learned to detect the salient objects at the scale at which it is trained  [i.e. (256, 256)]. So giving larger images as input definitely affects the output as it has never seen objects at that scale.

> I'm also wondering whether I can change the output image as a rectangle. (16:9) ratio.

Yes, you can definitely do so by resizing the input image's larger side to 256 (maintaining the aspect ratio). Pad the shorter side to make the input image of size (256, 256) and then feed it to the model. In fact, this is what I do (check the dataloader). You can finally crop out the padding on the output mask and resize the cropped mask to the shape of the original input image.

> I don't want to resize by some image processing tools because I think it will change the saliency value for each pixel and destroy the one-to-one corresponding pixel relationship.

By the method explained above you would get the closest possible results. Yes, the saliency value for each pixel would change but I guess that would be minimal and only be due to interpoation.

**Comment by sairajk** [comment]
Closing the issue, feel free to reopen.
